src/main/java/IntegerRangeInputValidator.java

```diff
@@ -0,0 +1,28 @@
+public class IntegerRangeInputValidator {
+    private int low;
+    private int high;
+
+    public IntegerRangeInputValidator(int low, int high) {
+        this.low = low;
+        this.high = high;
+    }
+
+    public boolean isValid(String input) {
+        if (!canParse(input)) return false;
+        if (isInRange(Integer.parseInt(input))) return true;
```

Comment:
It's interesting that `Integer.parseInt` ends up getting called twice here - any good ways you see to avoid it?